src/client/actions/TypingStart.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const Action = require('./Action');
+const { Events } = require('../../util/Constants');
+const textBasedChannelTypes = ['dm', 'text', 'news'];
+
+class TypingStart extends Action {
+  handle(data) {
+    const channel = this.getChannel(data);
+    let user;
+    if (data.guild_id) {
+      const guild = this.client.guilds.cache.get(data.guild_id);
+      if (guild) {
+        user = this.getMember(data.member, guild).user;
+      }
+    } else {
+      user = this.getUser(data);
+    }
```

Comment:
This is *a bit* of a duplicated code, can't we get a function that does it for us so we can re-use it?